syntax/indexmenu.php

```diff
@@ -769,10 +769,14 @@ public function parseNs($ns, $id = false)
         if ($ns == '..') {
             $ns = ":";
         }
-        $ns = "$ns:arandompagehere";
-        $resolver = new PageResolver($id);
-        $ns = getNs($resolver->resolveId($ns));
-        return $ns === false ? '' : $ns;
+        if (class_exists('dokuwiki\File\PageResolver')) {
+            $ns = "$ns:arandompagehere";
+            $resolver = new PageResolver($id);
+            $ns = getNs($resolver->resolveId($ns));
+            return $ns === false ? '' : $ns;
+        } else {
+            return resolve_id(getNS($id), $ns);
```

Comment:
In general `resolve_id()` should not be used. But I guess the old code did this already…

Reply:
Yes, the old code already uses this function. I noticed that there's a lot of depreciation errors about IndexMenu plug-in, as it still references "resolve_id" and "resolve_pageid" functions that wasn't migrated to PageResolver yet, so I believe that we migrate all those functions or keep it until we migrate them all.